aip/general/0123.md

```diff
@@ -78,6 +80,8 @@ message Topic {
 - Pattern variables **must** be the singular form of the resource type e.g. the
   pattern variable representing a `Topic` resource ID is named `{topic}`.
 - Pattern variables **must** conform to the format `[a-z][_a-z0-9]*[a-z0-9]`.
+- Singular **must** be the lower camel case of the type.
+- Plural **must** be the lower camel case plural of the singular.
```

Comment:
Can you also include (as sub bullets possibly) that the these **must** correspond to the the `resource_id` pattern variable and collectionId segments respectively? I actually thing the bullet on line 80 states this requirement but...perhaps we need to be more explicit in tying them to the annotation fields.

Reply:
I can say that for the resource_id, but the collection segments unfortunately I
can't: https://google.aip.dev/122#collection-identifiers.

I guess I could say it **must** except for nested collections, WDYT?

Reply:
Sorry I don't quite follow why not...

The `plural: publishers` must be the collection ID in resource pattern(s) like `projects/{project}/publishers/{publisher}.

When would they differ?

Reply:
for `userEvents` the resource name may look like
`user/{user}/events/{user_event}` (the example they have in the link above).

I'm not super happy with that pattern to be honest because of the inconsistency,
but it's been a hard change to agree on (the up-coming revisions AIP preferred
/book/*/revisions/ over /book/*/bookRevisions/)

Reply:
Oh I see, I see. So for now we don't mandate a correlation between the two, but we may end up doing just that if we can resolve those edge cases. Thanks!

Reply:
precisely! I do hope we can at some point.